common/bytes.go

```diff
@@ -105,8 +105,8 @@ func ParseHexOrString(str string) ([]byte, error) {
 }
 
 // RightPadBytes zero-pads slice to the right up to length l.
-func RightPadBytes(slice []byte, l int) []byte {
-	if l <= len(slice) {
+func RightPadBytes(slice []byte, l uint64) []byte {
+	if l <= uint64(len(slice)) {
```

Comment:
This change is not allowed because it is an API change. Also, there can be no overflow here because there is no addition, it's just a comparison between two values of type `int`. Please revert this.

Reply:
Myeah, ok. How about adding a second RightPadBytes method then, and deprecate this one?

Reply:
We don't even have to deprecate it (because the method itself doesn't do any unsafe conversion), just leave it as is

Reply:
When referring to the length of a slice, it is best to use type `int`, since this is also the type returned by `len(slice)`. So I think the type of `l` needs to be `int` here.

Reply:
I think not. Because that is what leads to bugs like [this](https://go-review.googlesource.com/c/go/+/393015/3/src/reflect/type.go#b642) and [this](https://github.com/ethereum/go-ethereum/security/advisories/GHSA-wjxw-gh3m-7pm5). 

It is _safe_ to convert `len<int>` into `len<uint64>`. Doing it the other way around forces the caller do an unsafe conversion (or, well, I mean the caller can add extra checks on all call sites, but that's not really realistic)